lib/resources/windows_task.rb

```diff
@@ -61,7 +61,99 @@ def task_to_run
     def run_as_user
       info[:run_as_user]
     end
+    
+    def hostname
```

Comment:
It seems like this might be a good candidate for dynamically defining these methods.